### ISSUE TITLE
linux wait4 syscall is wrong

### ISSUE THREAD

**Issue by PucklaJ** [odin-lang/Odin#3356]
## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System & Odin Version:
* Please paste `odin report` output:
```
Where to find more information and get into contact when you encounter a bug:

	Website: https://odin-lang.org
	GitHub:  https://github.com/odin-lang/Odin/issues


Useful information to add to a bug report:

	Odin:    dev-2024-03:3a8971c26
	OS:      Arch Linux, Linux 6.1.71-1-lts
	CPU:     Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz
	RAM:     31959 MiB
	Backend: LLVM 14.0.6
```
## Expected Behavior

The `wait4` syscall for x86_64 is defined as follows (according to [blog.rchapman.org](https://blog.rchapman.org/posts/Linux_System_Call_Table_for_x86_64/).
| %rax | System call | %rdi | %rsi | %rdx | %r10 |
| ----- | ----- | --- | --- | --- | --- |
| 61 | sys_wait4 | pid_t upid | int *stat_addr | int options | struct rusage *ru |

Other websites state the same. Therefore the procedure in `core:sys/linux` should have the same signature:
```odin
wait4 :: proc "contextless" (pid: Pid, status: ^u32, options: Wait_Options, usage: ^Usage_Struct)
```

## Current Behavior

Currently the wait4 procedure is defined as follows:
```odin
wait4 :: proc "contextless" (pid: Pid, status: ^u32, options: Wait_Options)
```

## Failure Information (for bugs)

This led to either a segfault or an error in the wait4 syscall because it's reading arbitrary data from the %r10 register.
When manually calling the syscall like that:
```odin
linux.syscall(linux.SYS_wait4, linux.Pid(-1), cast(rawptr)nil, cast(u32)0, cast(rawptr)nil)
```
it worked completely fine.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. use linux.fork to create a child process
2. call linux.execve to execute a command
3. call linux.wait4 in the parent process
4. It might return an error or might segfault or it might even run fine

### Failure Logs

Please include any relevant log snippets or files here.

**Comment by flysand7** [comment]
Thanks for the bug report, I'll make a PR to fix this tomorrow

**Comment by PucklaJ** [comment]
Fixed by #3429
